neo/rawio/axonarawio.py

```diff
@@ -0,0 +1,374 @@
+ data.bin raw data file
+ data.1 spike times and waveforms for tetrode 1, or stereotrodes 1 and 2
+ data.2 spikes times and waveforms for tetrode 2, or stereotrodes 3 and 4
+ …
+ data.32 spikes times and waveforms for tetrode 32
+ data.eeg continuous 250 Hz EEG signal, primary channel
+ data.eegX continuous 250 Hz EEG signal, secondary channel (X = 1..16)
+ data.egf high resolution 4800 Hz version of primary EEG channel
+ data.egfX high resolution 4800 Hz version of primary EEG channel (X = 1..16)
+ data.pos tracker position data
+ data.inp digital input and keypress timestamps
+ data.stm stimulation pulse timestamps
+ data.spk spikes times and waveforms for monotrodes (single electrodes) 1 to 16
+ data.epp field potential parameters
+ data.epw field potential waveforms
+ data.log DACQBASIC script optional user-defined output files
+
+Here we are only going to support .set and .bin files for now (raw cont. data). 
+ At least the following could be derived from it: .eeg, .egf, .X, .pos.
+
+Rules for creating a new class:
+  1. Step 1: Create the main class
+
+  2. Step 2: RawIO test:
+
+  3. Step 3 : Create the neo.io class with the wrapper
+    * Create a file in neo/io/ that ends with "io.py"
+    * Create a class that inherits both your RawIO class and BaseFromRaw class
+    * copy/paste from neo/io/exampleio.py
+
+  4.Step 4 : IO test
+    * create a file in neo/test/iotest with the same previous name with "test_" prefix
+    * copy/paste from neo/test/iotest/test_exampleio.py
+
+Author: Steffen Buergers
+
+"""
+
+from .baserawio import (BaseRawIO, _signal_channel_dtype, _signal_stream_dtype,
+                _spike_channel_dtype, _event_channel_dtype, _common_sig_characteristics)
+import numpy as np
+import os
+import re
+import contextlib
+import datetime
+
+
+class AxonaRawIO(BaseRawIO):
+    """
+    Class for reading raw, continuous data from the Axona dacqUSB system:
+    http://space-memory-navigation.org/DacqUSBFileFormats.pdf
+
+    The raw data is saved in .bin binary files with an accompanying
+    .set file about the recording setup (see the above manual for details).
+
+    Usage:
+        >>> import neo.rawio
+        >>> r = neo.rawio.AxonaRawIO(filename=os.path.join(dir_name, base_filename))
+        >>> r.parse_header()
+        >>> print(r)
+        >>> raw_chunk = r.get_analogsignal_chunk(block_index=0, seg_index=0,
+                            i_start=0, i_stop=1024,  channel_names=channel_names)
+        >>> float_chunk = reader.rescale_signal_raw_to_float(raw_chunk, dtype='float64',
+                            channel_indexes=[0, 3, 6])
+        >>> spike_timestamp = reader.spike_timestamps(spike_channel_index=0,
+                            t_start=None, t_stop=None)
+        >>> spike_times = reader.rescale_spike_timestamp(spike_timestamp, 'float64')
+        >>> ev_timestamps, _, ev_labels = reader.event_timestamps(event_channel_index=0)
+
+    """
+    # TODO Why do I need these?
```

Comment:
It can be good for auto guess the file format, even if the "bin" is quite fuzzy.

Reply:
In my opinion, the user could pass either a bin or a set file and the class should look for the companion file

Reply:
Yes, I agree. This attr is for auto guess the file format only anf is independant of the init.

Reply:
Ok, users can input `filename`, `filename.set` or `filename.bin`. 

I do not think `extensions = ['bin']` is used anywhere for auto-guessing, remove?